update-go-version.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Updates our various build files, CI configs, ... to the latest released go
+# version in a given minor release stream (1.x)
+# This requires yq and jq in addition to fairly standard shell tools (curl, grep, sed, ...)
+
+golang_base_version=$1
+latest_version=$(curl --silent  'https://golang.org/dl/?mode=json&include=all' | jq -r '.[].files[].version'  |uniq | sed -e 's/go//' |sort -V |grep ${golang_base_version}|tail -1)
+echo "Updating golang version to $latest_version"
+
+go mod edit -go ${golang_base_version}
+sed -i "s/^stack: go 1\...\$/stack: go ${golang_base_version}/" appveyor.yml
+sed -i "s,^FROM registry.svc.ci.openshift.org/openshift/release:golang-1\... AS builder\$,FROM registry.svc.ci.openshift.org/openshift/release:golang-${golang_base_version} AS builder," images/openshift-ci/Dockerfile
+sed -i "s/GOVERSION: .*\$/GOVERSION: \"${latest_version}\"/" .circleci/config.yml
+sed -i "s/^GO_VERSION=.*$/GO_VERSION=${latest_version}/" centos_ci.sh
```

Comment:
@cfergeau except centos ci all other CI config is in yaml, shouldn't we use `yq` for those instead `sed`?

Reply:
Yup, good point. For circle-ci, I don't know how to do it cleanly, I tried `yq write --inplace ./.circleci/config.yml 'jobs[*].environment.(GOVERSION==*).GOVERSION' "${latest_version}"` but this adds empty `environment{}` fields in jobs which don't have it.

Reply:
@cfergeau what if we do it in 2 steps instead single line?
```
$ yq write --inplace ./.circleci/config.yml 'jobs.build.environment.GOVERSION' "${latest_version}"
$ yq write --inplace ./.circleci/config.yml 'jobs.build_mock_release.environment.GOVERSION' "${latest_version}"
```

Reply:
This will break as soon as we rename one of these, or add a new one.